### PR TITLE
bug :: preserve original behavior for getStringParameter

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

**Review by jpdahlke — DISMISSED** [review]
The two situations we were running into:
- getStringParameter was no longer returning "" when the value was "" (nor was the new method getParameterAsConcatString)
- the new getParameterAsString method was returning null for single entry parameters with a value of "".

This looks to solve both of those and add tests to show it.

**Comment by jpdahlke** [comment]
This will be included in a patch release of 8.19.1

File: src/test/java/emissary/core/BaseDataObjectTest.java [hunk 1]
```diff
@@ -813,8 +831,14 @@ void testPutParametersWithPolicy() {
         this.b.putParameters(map, IBaseDataObject.MergePolicy.KEEP_ALL);
 
         assertEquals("uno;uno", this.b.getStringParameter("ONE"), "When specifying KEEP_ALL values must all stay");
+        assertEquals("uno;uno", this.b.getParameterAsConcatString("ONE"), "When specifying KEEP_ALL values must all stay");
+        assertEquals("uno", this.b.getParameterAsString("ONE"), "When specifying KEEP_ALL values must all stay");
```
Comment (fbruton) [inline src/test/java/emissary/core/BaseDataObjectTest.java:835]: This comment should be updated to state that the first value is retained.

File: src/test/java/emissary/core/BaseDataObjectTest.java [hunk 2]
```diff
@@ -813,8 +831,14 @@ void testPutParametersWithPolicy() {
...
         assertEquals("deux;dos", this.b.getStringParameter("TWO"), "When specifying KEEP_ALL values must all stay");
+        assertEquals("deux;dos", this.b.getParameterAsConcatString("TWO"), "When specifying KEEP_ALL values must all stay");
+        assertEquals("deux", this.b.getParameterAsString("TWO"), "When specifying KEEP_ALL values must all stay");
```
Comment (fbruton) [inline src/test/java/emissary/core/BaseDataObjectTest.java:838]: This comment should be updated to state that the first value is retained.

File: src/test/java/emissary/core/BaseDataObjectTest.java [hunk 3]
```diff
@@ -895,6 +931,8 @@ void testAppendDuplicateParameters() {
         this.b.appendParameter("YO", "GABBA");
         this.b.appendParameter("YO", "GABBA");
         assertEquals("GABBA;GABBA", this.b.getStringParameter("YO"), "Appended duplicate parameters should be preserved");
+        assertEquals("GABBA;GABBA", this.b.getParameterAsConcatString("YO"), "Appended duplicate parameters should be preserved");
+        assertEquals("GABBA", this.b.getParameterAsString("YO"), "Appended duplicate parameters should be preserved");
```
Comment (fbruton) [inline src/test/java/emissary/core/BaseDataObjectTest.java:935]: Update comment

File: src/test/java/emissary/core/BaseDataObjectTest.java
Comment (fbruton) [inline src/test/java/emissary/core/BaseDataObjectTest.java:1]: Update `getParameterAsString` assert messages to match the expected behavior.

**Review by fbruton — CHANGES_REQUESTED** [review]

**Review by jpdahlke — APPROVED** [review]

**Review by fbruton — APPROVED** [review]

**Pull request merged**